scipy/ndimage/_measurements.py

```diff
@@ -630,6 +630,10 @@ def single_group(vals):
     if labels is None:
         return single_group(input)
 
+    # manually cast to numpy
+    # since libaries (e.g. dask) that implement __array_function__
+    # will not return a numpy array from broadcast_arrays
+    labels = np.asarray(labels)
```

Comment:
Could you use xp.broadcast_arrays instead?

Reply:
Not sure I follow here.

This module operates on ndarrays only, so we shouldn't use array API here.

Reply:
This function is not array api compliant.
It used to accidentally work before with non-numpy backends because `np.broadcast_arrays` would quietly call `np.asarray`. Dask however has a NEP18 hook.
This explicit cast is fine until there is proper array API compliance for the function, which is definitely out of scope for this PR.